com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Watchpoint.java

```
@@ -0,0 +1,252 @@
+                if (!watchpointRequests.isEmpty() && !future.isDone()) {
+                    this.putProperty("verified", true);
+                    future.complete(this);
+                }
+            });
+        subscriptions.add(subscription);
+
+        List<EventRequest> watchpointRequests = new ArrayList<>();
+        List<ReferenceType> types = vm.classesByName(className);
+        for (ReferenceType type : types) {
+            watchpointRequests.addAll(createWatchpointRequests(type));
+        }
+
+        requests.addAll(watchpointRequests);
+        if (!watchpointRequests.isEmpty() && !future.isDone()) {
+            this.putProperty("verified", true);
+            future.complete(this);
+        }
+
+        return future;
+    }
+
+    private List<WatchpointRequest> createWatchpointRequests(ReferenceType type) {
+        List<WatchpointRequest> watchpointRequests = new ArrayList<>();
+        Field field = type.fieldByName(fieldName);
+        if (field != null) {
+            if ("read".equals(accessType)) {
+                watchpointRequests.add(vm.eventRequestManager().createAccessWatchpointRequest(field));
+            } else if ("readWrite".equals(accessType)) {
+                watchpointRequests.add(vm.eventRequestManager().createAccessWatchpointRequest(field));
+                watchpointRequests.add(vm.eventRequestManager().createModificationWatchpointRequest(field));
+            } else {
+                watchpointRequests.add(vm.eventRequestManager().createModificationWatchpointRequest(field));
+            }
+        }
+
+        watchpointRequests.forEach(request -> {
+            request.setSuspendPolicy(WatchpointRequest.SUSPEND_EVENT_THREAD);
+            if (hitCount > 0) {
+                request.addCountFilter(hitCount);
+            }
+            request.enable();
+        });
+        return watchpointRequests;
+    }
+
+    @Override
+    public String getLogMessage() {
+        return null;
```

throw UnsupportedException  rather than null/do nothing logic